src/renderer/src/pages/settings/settings-general.tsx

```diff
@@ -79,28 +102,40 @@ export function SettingsGeneral({
         </Button>
       </div>
 
+      <h3>{t("language")}</h3>
+      <>
+        <Select value={form.language} onChange={handleLanguageChange}>
+          {Object.keys(languageResources).map((language) => (
```

Comment:
Could you move this to a useState? This way you can also sort the languageOptions. Something like this:

![image](https://github.com/hydralauncher/hydra/assets/167933696/075c69b9-5dda-424b-8140-5ee7392ab5d0)

Reply:
Another thing that I noticed it that, the current string for portuguese language is "pt-BR", so the Select component won't have the correct language selected by default for who has portuguese language. I'm not sure the best way to fix this. Maybe something using the startsWith function. If you have an ideia let us know

Reply:
where did you see it is `pt-br`? could you send a pint of the file that sets this or file name and line? because where i looked also used the `import * as languageResources from "@locales";`, and there is `pt`

Reply:
> Could you move this to a useState? This way you can also sort the languageOptions. Something like this:
> 
> ![image](https://github.com/hydralauncher/hydra/assets/167933696/075c69b9-5dda-424b-8140-5ee7392ab5d0)
> 

makes sense to sort if the languages wil be shown as native name tho? the alfaphabets are not the same and it is not guaranteed that the lang codes sorted, will keep the native names sorted. i can sort after calling the toNativeName tho

Reply:
one thing that i was thinking is, if there are a LOT of languages available, the select will be massive, and i have not tested a scroll there. one option is a maxheight and scroll, another option is to use smt like a select2 with a max shown opts

Reply:
I think a scroll should be fine, could you try it?
And yes, the sort should happen after you get the native names

Reply:
the thing is, i will have ti sort by native name and covert back tolang code... will i think there is a good way to do this, will try here. will also try the scroll

Reply:
> Another thing that I noticed it that, the current string for portuguese language is "pt-BR", so the Select component won't have the correct language selected by default for who has portuguese language. I'm not sure the best way to fix this. Maybe something using the startsWith function. If you have an ideia let us know

did you found where says the stribg is `pt-BR` @zamitto?

Reply:
you can use a clean install to test and see
![image](https://github.com/hydralauncher/hydra/assets/167933696/d8ff617a-67a7-4074-918f-e6d929a1a9d7)

Reply:
only way i can think to fix it is to convert the `pt-BR` to `pt` with split before trying to set the default, as we dont have pt-PT anyway, but that seems too "gambit" to me...

sorry, crazy week at work, only could look at it now

Reply:
lol, seems that you have done just that hahah

![image](https://github.com/hydralauncher/hydra/assets/16529503/dffaee0b-c6eb-4986-bc4b-0424ce236aae)

well, if you dont see a problem with it, them LGTM :)